ouroboros-consensus/docs/Contributing.md

```diff
@@ -1,5 +1,35 @@
-TODO eg git history goals etc
+# How to contribute to the Consensus Layer
 
-See the
-https://github.com/input-output-hk/ouroboros-network/ouroboros-consensus/docs/StyleGuide.md
-document.
+See the follow documents.
+
+  * this repository's [README][repo-README]; it links to design documents
```

Comment:
To be frank, we have so MANY rules upfront that new contributor has to adhere to, it's a bit overwhelming for new comers.
It's not a comment to this PR, it's a more general observation.

Reply:
The concrete rules that come to mind are code formatting and the git process. Is that all of what you're referring to here? Am overlooking something else? I want to make sure we're talking about the same thing.

My current thought is that I don't see an alternative for this kind of project. If our plan were move fast, fail fast, iterate quickly to discover customer preferences etc, and so on, then it'd be different. Instead, the keyword I picked up on from transition discussions with Thomas and Edsko along these lines is _we're deliberate_. A structured approach seems appropriate for that.

My motivation for this PR (and the recent similar ones) is that, if we have rules, we should write them down. That helps both PR authors and reviewers approach the topic.

But I agree that the enumerated rules seem intimidating/overwhelming when collated like this. I think one actionable thing your comment here has prompted me to think about is explicitly explaining our expectations around enforcement of these rules. Basically "this is a list of things we will be looking for when we review PRs", "we don't expect PRs to start perfect", "you'll eventually internalize these", etc, etc.

Does that sound like a reasonable why to address the sentiment you shared in your comment here? Of course we can chat about this on our next call if you'd prefer.

Reply:
> the keyword I picked up on from transition discussions with Thomas and Edsko along these lines is we're deliberate. A structured approach seems appropriate for that.

Yeah, I understand the intent, I was just referring to burned of someone who would like to do a small PR to our codebase but would be stopped by "the process".

> I think one actionable thing your comment here has prompted me to think about is explicitly explaining our expectations around enforcement of these rules. Basically "this is a list of things we will be looking for when we review PRs", "we don't expect PRs to start perfect", "you'll eventually internalize these", etc, etc.

👍  totally agree with that one, we can put that upfront in the `Contributing.md`. I think it does the issue directly, we keep our structured approach but we don't scare of newcommers.